File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Program "m" as the height
    // State that m can't be less than 0 or more than 23
    int m;
    do
    {
        m = get_int("height: ");
    }
    while (m < 0 || m > 23);
    // Define "i" as representing the line number
    // Define "s" as the amount of spaces each line has
    // Define "h" as the amount of hashes each line has
    for (int i = 0; i < m; i++)
    {
        for (int s = 0; s < m - 1 - i; s++)
        {
            printf(" ");
        }
        for (int h = 0; h < i + 2; h++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: mario.py
from cs50 import get_int


def main():
    # Prompt user for height (m) until a value between 0 and 23 is given
    while True:
        m = get_int('height: ')
        if m >= 0 and m <= 23:
            break

    # Calculate pyramid's width (height + 1)
    w = m + 1
    # Iterate over pyramid's rows
    for i in range(m):
        # Iterate over pyramid's columns
        for j in range(w):
            # Print spaces so that top row of pyramid has two hashes
            if (j < w - 2 - i):
                print(' ', end='')
            # Print hashes
            else:
                print('#', end='')
        # End line and go to next line until height is reached
        print()


if __name__ == '__main__':
    main()
